Improve fan behaviour for GL.iNet MT3000 router

lukasz1992 opened the pull request:
- double the number of cooling-levels one can use
- improve the fan behaviour:
  - turn on fan on lower temperatures, but make it quiet
  - increase RPMs faster on higher temperatures
  - reduce probability of constant start-stop actions

This is an improvement for all MT7981 devices with pwm fan

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

dangowrt reviewed (commented):
Are you sure the fan is even rotating at all at 25% PWM duty level? If not it would become a heating rather than a cooling device...

lukasz1992 commented:
@dangowrt Yes - 13% is being required to make it start (34).

OEM dts sets the first speed to 14% (36 / 256).
63 should be more than safe.

lukasz1992 commented:
bump.

Tested openwrt 23.05.0 - device is getting hot and aggressively stops/starts fan

hauke commented:
This pull request has formal issues:

> === Checking commit '64b762a8457c73da78ebdf147a39a176546b6079'
> Pull request should not include merge commits
> Author name (Lukasz M) seems ok
> Commit subject line MUST start with '<area>: ' (Merge branch 'main' into improve-mt3000-fan)
> Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: Lukasz M > <13104955+lukasz1992@users.noreply.github.com>')
> 
> A commit message exists
> === Checking commit '95167f1ff43a98c3fbf0d83a7cc50f764a12826f'
> Pull request should not include merge commits
> Author name (Lukasz M) seems ok
> Commit subject line MUST start with '<area>: ' (Merge branch 'main' into improve-mt3000-fan)
> Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: Lukasz M <13104955+lukasz1992@users.noreply.github.com>')
> 
> A commit message exists
> === Checking commit 'd604c4660bc5736a45913416d78d4941f81a07ac'
> Author name (Łukasz M) seems ok
> Commit subject line MUST start with '<area>: ' (Improve the fan behaviour:   - turn on also on lower temperatures   - increase RPMs faster on higher temperatures   - reduce probability of constant start-stop actions With this commit, device is no longer hot in room temperature)
> Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: Łukasz M <lukasz1992@example.com>')
> 
> A commit message exists
> Error: Process completed with exit code 1.

dangowrt commented:
You have to rebase and force-push to avoid the creation of merge-commits.

lukasz1992 commented:
I will do it until tomorrow, thanks

herbetom commented:
You're still missing the `Signed-off-by: John Doe <mail@example.org>` in the commit message @hauke mentioned as well as a properly formated Commit Message. This is not the be all end end all, but something along the lines of:


```
mediatek: mt7981: improve fan behaviour

This doubles the number of cooling-levels. In addition
the fan is turned on with a low speed at lower temperatures
and with a higher speed at higher temperatures.
This also attempts to reduce the likelihood of 
constant start-stop actions.

The change only affects the GL.iNet MT3000 and has been tested with it.

Signed-off-by: John Doe <mail@example.org>
```

Feel free to use as much or as little as you would like. The commit message can be edited with `git commit --amend`.

lukasz1992 committed:
mediatek: mt7981: improve fan behaviour

This doubles the number of cooling-levels.
In addition the fan is turned on with a low speed at lower temperatures
and with a higher speed at higher temperatures.
This also attempts to reduce the likelihood of constant start-stop actions.

The change only affects the GL.iNet MT3000 and has been tested with it.

Signed-off-by: Łukasz M <lukasz1992m@gmail.com>

lukasz1992 commented:
@herbetom @dangowrt done :)

blocktrron commented:
Merged to my staging tree - thanks!
